src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.Utf8JsonStreamReader.cs

```diff
@@ -0,0 +1,1801 @@
+        private static readonly byte[] ImportsPropertyName = Encoding.UTF8.GetBytes("imports");
+        private static readonly byte[] RuntimeIdentifierGraphPathPropertyName = Encoding.UTF8.GetBytes("runtimeIdentifierGraphPath");
+        private static readonly byte[] WarnPropertyName = Encoding.UTF8.GetBytes("warn");
+        private static readonly byte[] IconUrlPropertyName = Encoding.UTF8.GetBytes("iconUrl");
+        private static readonly byte[] LicenseUrlPropertyName = Encoding.UTF8.GetBytes("licenseUrl");
+        private static readonly byte[] OwnersPropertyName = Encoding.UTF8.GetBytes("owners");
+        private static readonly byte[] PackageTypePropertyName = Encoding.UTF8.GetBytes("packageType");
+        private static readonly byte[] ProjectUrlPropertyName = Encoding.UTF8.GetBytes("projectUrl");
+        private static readonly byte[] ReleaseNotesPropertyName = Encoding.UTF8.GetBytes("releaseNotes");
+        private static readonly byte[] RequireLicenseAcceptancePropertyName = Encoding.UTF8.GetBytes("requireLicenseAcceptance");
+        private static readonly byte[] SummaryPropertyName = Encoding.UTF8.GetBytes("summary");
+        private static readonly byte[] TagsPropertyName = Encoding.UTF8.GetBytes("tags");
+        private static readonly byte[] MappingsPropertyName = Encoding.UTF8.GetBytes("mappings");
+        private static readonly byte[] HashTagImportPropertyName = Encoding.UTF8.GetBytes("#import");
+        private static readonly byte[] ProjectReferencesPropertyName = Encoding.UTF8.GetBytes("projectReferences");
+        private static readonly byte[] EmptyStringPropertyName = Encoding.UTF8.GetBytes(string.Empty);
+
+        internal static PackageSpec GetPackageSpecUtf8JsonStreamReader(Stream stream, string name, string packageSpecPath, string snapshotValue)
+        {
+            var reader = new Utf8JsonStreamReader(stream);
+            PackageSpec packageSpec;
+            packageSpec = GetPackageSpec(ref reader, name, packageSpecPath, snapshotValue);
+
+            if (!string.IsNullOrEmpty(name))
+            {
+                packageSpec.Name = name;
+                if (!string.IsNullOrEmpty(packageSpecPath))
+                {
+                    packageSpec.FilePath = Path.GetFullPath(packageSpecPath);
+
+                }
+            }
+            return packageSpec;
+        }
+
+        internal static PackageSpec GetPackageSpec(ref Utf8JsonStreamReader jsonReader, string name, string packageSpecPath, string snapshotValue)
+        {
+            var packageSpec = new PackageSpec();
+
+            List<CompatibilityProfile> compatibilityProfiles = null;
+            List<RuntimeDescription> runtimeDescriptions = null;
+            var wasPackOptionsSet = false;
+            var isMappingsNull = false;
+            string filePath = name == null ? null : Path.GetFullPath(packageSpecPath);
+
+            if (jsonReader.TokenType == JsonTokenType.StartObject)
```

Comment:
It appears that if the file has a semantic error, for example, it's a start array, or just some literal value, this code silently ignores the error. I'm not sure what the Newtonsoft.Json code does, but typically deserializers will throw an exception.

There are so many tests, it's hard to find if such a test already exists. Do you know if there are error condition tests?

Reply:
There wasn't a test for this specific case, I just tried one and it fails silently in both versions of this code. I'm happy to add exceptions and stop failing silently but that would be a different behavior from what we already do.

Reply:
ok, I don't want to add scope creep, so even though I feel like this behaviour is wrong, given that assets and dgspec files are only expected to be written by nuget, which will always output a valid file, it's unlikely that a customer will have an invalid file being read.

We haven't had any customer complains in the years since the Newtonsoft.Json code was changed in this way, so consider it out of scope for this STJ work.